highway.tex

```diff
@@ -24,807 +26,1323 @@
+For a block $B$, we refer to all the blocks that are on the other branches as \emph{competing} with $B$, as if any of them would be chosen, $B$ could not.
 
-In section \ref{sectionLiveness}, we introduce additional rules that guarantee not only safety, but also liveness; decisions can be relied on and are eventually actually made.
 
-Finally, section \ref{sectionPermissionless} puts the abstract consensus algorithm in the context of a distributed proof of stake network, outlines how deposits, rewards and penalties can be handled, and addresses the problem of long range attacks.
+\subsection{Practical challenges}
 
-\newpage
+{\bf Strong optimistic finality.} While since the initial definition of the partially synchronous model by Dwork, Lynch and Stockmeyer~\cite{DLS88} the vast body of research was created to optimize various parameters of protocols in this setting, most of it was written under the semi-formal assumption that the existence of more than $n/3$ dishonest nodes predates the existence of any provable guarantees for such protocols.
+Such an assumption stems from the fact that, as proven in the original paper, it is not possible for a partially synchronous protocol to guarantee both liveness and finality if $n<3f+1$.
 
+In spite of this negative result, it is however possible to provide additional finality guarantees in the scenario where for a prolonged period even dishonest nodes do not deviate from the protocol and actively work to finalize blocks.
+Although it may not seem like an important observation from the perspective of classical security models, we stress that it carries significant practical consequences -- in most cases of blockchain deployments it may be assumed that most of the time basically all the nodes will actively collaborate to achieve consensus\footnote{As most systems rewards and punishes validators based on their behavior, offline nodes and small-scale attacks (ones that do not succeed in reverting a finalized block) are strongly disincentivised and hence not common.}.
+It is hence possible to provide much stronger finality guarantees for blocks created during such periods, so to revert them much more than $n/3$ validators would have to collude.
 
-\section{Messages and Finality}
-\label{sectionMessages}
 
+{\bf Flexibility.}
+Once the classical $n<3f+1$ bound is left aside, a natural tradeoff between finality and liveness occurs -- the stronger finality guarantee we require, the more validators need to honestly collaborate to finalize the block with such a guarantee.
+The tradeoff could be resolved by all the validators agreeing on a common finality threshold that they intend to use, but such solution would have significant limitations.
 
-\subsection{Justifications}
+In Highway, however, there is no need to agree upon a common threshold, so every validator is able to use a different one, or even several different thresholds.
```

Comment:
I think in a permissioned setting, a _validator_ doesn't even have to pick any threshold at all, and every _observer_ can choose theirs.